### PR TITLE
fix "podman run port forward range" flake

### LINKED ISSUE #21561: podman run port forward range: ncat: connection refused
```
[+1234s] not ok 512 [500] podman run port forward range
...
<+015ms> # # podman-remote --url unix:///var/tmp/podman_tmp_YncV exec -d c95eaa0d5b1359c76cb1f267a9c3ba726564b0414b34122723c631b38e0d1c7a nc -l -p 5866 -e /bin/cat
<+090ms> # c0a0d47f72158daa1fa8d61cc9882ca40623d31ceabee77f0ea4d0bf822238b8
         # #/vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv
         # #|     FAIL: ncat got data back (netmode=bridge port=5866)
         # #| expected: '9PD2bgUX1v'
         # #|   actual: 'Ncat: Connection refused.'
         # #\^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Only remote, and only f38. "Oh, f38, this is a CNI bug, and we don't care". Nope - still happening even after we bumped f38 VMs to netavark.

* fedora-38 : sys remote fedora-38 root host boltdb [remote]
  * PR #21512
    * [02-05 08:54](https://api.cirrus-ci.com/v1/artifact/task/6187533038518272/html/sys-remote-fedora-38-root-host-boltdb.log.html#t--00512) in [sys] [500] podman run port forward range
  * PR #21431
    * [02-06 06:45](https://api.cirrus-ci.com/v1/artifact/task/4753038120845312/html/sys-remote-fedora-38-root-host-boltdb.log.html#t--00512) in [sys] [500] podman run port forward range
  * PR #21413
    * [02-02 15:44](https://api.cirrus-ci.com/v1/artifact/task/6600965382471680/html/sys-remote-fedora-38-root-host-boltdb.log.html#t--00512) in [sys] [500] podman run port forward range
  * PR #21384
    * [01-31 10:08](https://api.cirrus-ci.com/v1/artifact/task/5499119775514624/html/sys-remote-fedora-38-root-host-boltdb.log.html#t--00511) in [sys] [500] podman run port forward range
  * PR #21366
    * [01-27 08:45](https://api.cirrus-ci.com/v1/artifact/task/6434932650672128/html/sys-remote-fedora-38-root-host-boltdb.log.html#t--00511) in [sys] [500] podman run port forward range
  * PR #17831
    * [02-06 11:13](https://api.cirrus-ci.com/v1/artifact/task/4634216978186240/html/sys-remote-fedora-38-root-host-boltdb.log.html#t--00512) in [sys] [500] podman run port forward range
    * [01-31 11:32](https://api.cirrus-ci.com/v1/artifact/task/5084308143079424/html/sys-remote-fedora-38-root-host-boltdb.log.html#t--00511) in [sys] [500] podman run port forward range

| x | x | x | x | x | x |
| ---: | ---: | ---: | ---: | ---: | ---: |
| sys(7) | remote(7) | fedora-38(7) | root(7) | host(7) | boltdb(7) |

### PR DESCRIPTION
**Pull request by Luap99**
For some reason this starting to flake f38. I don't think the issue in podman rather the test start nc -l in the background so it may not yet have bound the port in the container when we try to connect. To fix this simply add some retry logic to nc.

While at it also add pasta to this test and make it use defer-assertion-failures to run all loop iterations before reporting the errors.

Fixes #21561 (hopefully)

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by Luap99** [comment]
@edsantiago PTAL

File: test/system/500-networking.bats [hunk 1]
```diff
@@ -729,10 +738,22 @@ nameserver 8.8.8.8" "nameserver order is correct"
...
-            # change this are not portable across distros. -w seems to work.
-            run nc -w 1 127.0.0.1 $port <<<$random
+
+            # we have to rety ncat as it can flake as we exec in the background so nc -l
+            # might not have bound the port yet, retry seems simpler than checking if the
+            # port is bound in the container, https://github.com/containers/podman/issues/21561.
+            retries=5
+            while [[ $retries -gt 0 ]]; do
+                # -w 1 adds a 1 second timeout. For some reason, ubuntu's ncat
+                # doesn't close the connection on EOF, and other options to
```
Comment (edsantiago) [inline test/system/500-networking.bats:748]: I wonder if this is still relevant, given that we no longer test on ubuntu? That special-case code dates from March 2022, #13646

**Review by edsantiago — APPROVED** [review]
LGTM, thank you

File: test/system/500-networking.bats (hunk 1 above)
Comment (Luap99) [reply test/system/500-networking.bats:748]: We would need to test that, although it makes sure nc would time out and not hang forever.

**Pull request merged**